WalletWasabi/Blockchain/Keys/KeyManager.cs

```diff
@@ -541,9 +541,9 @@ public IEnumerable<HdPubKey> GetKeys(KeyState? keyState = null, bool? isInternal
 			}
 		}
 
-		public int CountConsecutiveCleanKeys(bool isInternal)
+		public int CountConsecutiveUnusedKeys(bool isInternal)
 		{
-			var keyIndexes = GetKeys(KeyState.Clean, isInternal).Select(x => x.Index).ToArray();
+			var keyIndexes = GetKeys(x => x.IsInternal == isInternal && x.KeyState != KeyState.Used).Select(x => x.Index).OrderBy(x => x).ToArray();
```

Comment:
@nopara73 We don't need `OrderBy()` at all.
```suggestion
			var keyIndexes = GetKeys(x => x.IsInternal == isInternal && x.KeyState != KeyState.Used).Select(x => x.Index).ToArray();
```